instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java

```diff
@@ -101,17 +100,6 @@ public Resource createResource(ConfigProperties config) {
         .orElseGet(Resource::empty);
   }
 
-  @Override
-  public boolean shouldApply(ConfigProperties config, Resource resource) {
```

Comment:
this check needs to be removed, otherwise this detector won't run if we already found a name in `SpringBootServiceVersionDetector`

Reply:
the same optimization is added again with https://github.com/open-telemetry/opentelemetry-java/pull/6222

Reply:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10523/files#r1486170239 comment from @laurit does not seem to support this. So, I would suggest to not remove `implements ConditionalResourceProvider` in this PR, and discuss  in another place if we should keep `shouldApply` or not, to be able to go ahead with this PR.

Reply:
you're right - I got confused by this one